patches/minecraft/net/minecraft/entity/Entity.java.patch

```diff
@@ -222,15 +222,27 @@
     }
  
     public boolean func_70028_i(Entity p_70028_1_) {
-@@ -2046,6 +2073,7 @@
+@@ -2046,6 +2073,11 @@
  
     @Nullable
     public Entity func_212321_a(DimensionType p_212321_1_) {
++      return this.changeDimension(p_212321_1_, func_184102_h().func_71218_a(p_212321_1_).func_85176_s());
++   }
++   @Nullable
++   public Entity changeDimension(DimensionType p_212321_1_, net.minecraftforge.common.util.ITeleporter teleporter) {
 +      if (!net.minecraftforge.common.ForgeHooks.onTravelToDimension(this, p_212321_1_)) return null;
        if (!this.field_70170_p.field_72995_K && !this.field_70128_L) {
           this.field_70170_p.func_217381_Z().func_76320_a("changeDimension");
           MinecraftServer minecraftserver = this.func_184102_h();
-@@ -2063,16 +2091,9 @@
+@@ -2055,6 +2087,7 @@
+          this.field_71093_bK = p_212321_1_;
+          this.func_213319_R();
+          this.field_70170_p.func_217381_Z().func_76320_a("reposition");
++         Entity transportedEntity = teleporter.placeEntity(this, serverworld, serverworld1, this.field_70177_z, spawnPortal -> {
```

Comment:
Please add //Forge: Start/End vanilla logic to the braces so it's easier for us to grok the logic without the indentations.

Reply:
Done